### ISSUE TITLE
Number of parameters of a dense layer with a 2D input

### ISSUE THREAD

**Issue by ivanlen** [keras-team/keras#11866]
I am using 2D data in a classification problem using keras.
So I am defining a keras model as following:

```python
in_ = Input((5, 10))
out = Dense(100, activation='relu', name = 'dense_1')(in_)

model = Model(in_, out)
model.compile(loss='categorical_crossentropy', optimizer='adam')
model.summary() 
```

which returns a compiled model with the following number of parameters parameters

```python
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
input_9 (InputLayer)         (None, 5, 10)             0         
_________________________________________________________________
dense_1 (Dense)              (None, 5, 100)            1100      
=================================================================
Total params: 1,100
Trainable params: 1,100
Non-trainable params: 0
_________________________________________________________________
```
Reading the documentation and inspecting the [code](https://github.com/keras-team/keras/blob/88af7d0c97497b5c3a198ee9416b2accfbc72c36/keras/layers/core.py#L880) if the input shape has more than 1D it is going to use the last dimension as input dimension. 
Thus the layer has 1100 parameters = 10*100+ 100 (weights + bias).

What I don't understand is how the dot product between the input and the weights is performed. Because if the data is 2D with 5\*10 data points, and I am using only 100\*10 weights how are the weights and input values multiplied? If I change the input data from (5,10) --> (500,10) something changes?

I always have the idea that the output of the neuron k was something like:
`f_k = Relu(w_{k,1}*x_1 + w_k2*x_2 + ... + w_{k,n} * x_n + b_k)`
where the x_1, x_2 are ALL the previous outpus, however in this case that is 2D I don't know what is happening here.

**Comment by Golbstein** [comment]
It'll refer to "5" as batch.. What's the shape of your labels? Maybe you should consider adding `Flatten()` layer after the input.

**Comment by ivanlen** [comment]
> It'll refer to "5" as batch.. What's the shape of your labels? Maybe you should consider adding `Flatten()` layer after the input.

Supposing my data were a time series with 5 features and 10 timesteps, all my 5 different features would be multiplied by the same weights, right?

On the other side this were just the first layers of the network, then I a `Flatten()` layer, thanks for the suggestion btw.
